src/components/Dao/DaoProposalsPage.tsx

```diff
@@ -0,0 +1,152 @@
+import ProposalRow from "components/Proposal/ProposalRow";
+import { RouteComponentProps } from "react-router-dom";
+import { first } from "rxjs/operators";
+import { isAddress } from "lib/util";
+import moment from "moment-timezone";
+
+const PAGE_SIZE = 50;
+
+type SubscriptionData = Array<Proposal>;
+type IProps = IExternalProps & ISubscriptionProps<SubscriptionData>;
+type IExternalProps = {
+  daoState: IDAOState;
+  currentAccountAddress: Address;
+} & RouteComponentProps<any>;
+
+type OrderDirection = "asc" | "desc";
+
+
+const proposalsQuery = (dao: IDAOState, skip: number, titleSearch?: string): Observable<Array<Proposal>> => {
+  const filter: any = {
+  };
+
+  let orderDirection: OrderDirection = "asc";
+
+  if (!titleSearch) {
+    if (skip % PAGE_SIZE === 0) {
+      filter["closingAt_gt"] = moment().unix();
+    } else {
+      skip -= dao.numberOfQueuedProposals + dao.numberOfPreBoostedProposals + dao.numberOfBoostedProposals;
+      filter["closingAt_lt"] = moment().unix();
+    }
+  }
+
+  if (titleSearch?.trim()) {
+    orderDirection = "desc";
+    if (isAddress(titleSearch)) {
+      filter["proposer"] = titleSearch;
+    } else {
+      filter["title_contains"] = titleSearch;
+    }
+  }
+
+  return dao.dao.proposals({
+    where: filter,
+    orderBy: "closingAt",
+    orderDirection: orderDirection,
+    first: titleSearch ? undefined : PAGE_SIZE, // TEMPORARY UNTIL WE PASS "titleSearch" in line 143
+    skip,
+  }, { fetchAllData: true });
+};
+
+const DaoProposalsPage = (props: IProps) => {
+  const { data, hasMoreToLoad, fetchMore, daoState } = props;
+  const [filtering, setFiltering] = React.useState(false);
+  const [filterString, setFilterString] = React.useState("");
+  const [filteredProposalSet, setFilteredProposalSet] = React.useState(null);
+
+  const onSearchExecute = async (e: any) => {
+    let foundProposals: Array<Proposal>;
+    if ((e.type === "blur") || (e.key === "Enter")) {
+      if (filterString?.length) {
+        setFiltering(true);
+        foundProposals = await proposalsQuery(props.daoState, 0, filterString).pipe(first()).toPromise();
+      }
+      else {
+        foundProposals = null;
+      }
+      setFilteredProposalSet(foundProposals);
+      setFiltering(false);
+    }
+  };
+
+
+
+  const proposals = (filteredProposalSet ?? data).map((proposal: Proposal) => {
+    return <ProposalRow key={proposal.id} data={proposal} history={props.history} />;
+  });
+
+  return (
+    <div className={css.wrapper}>
+      <BreadcrumbsItem to={`/dao/${daoState.address}`}>Proposals</BreadcrumbsItem>
+      <div className={css.topBarWrapper}>
+        <div className={css.top}>
+          <h1 className={css.title}>Proposals</h1>
+          <div
+            // Temporary until https://github.com/daostack/alchemy/issues/2292 is ready.
+            style={{ pointerEvents: "none", backgroundColor: "gray"}}
+            className={css.createProposalButton}
+            data-test-id="createProposal">
```

Comment:
Why do we need this?

Reply:
You mean the grey button? It's temporary until your PR will be ready and we can open the new modal from this button.